### ISSUE TITLE
Song Editor: incorrect display when Reopen last project on start

### ISSUE THREAD

**Issue by midi-pascal** [LMMS/lmms#2424]
Master branch, Ubuntu 14.04 x86-64
When "Reopen last project on start" is checked in the settings, the project loads correctly on startup but the Song Editor displays only the first few bars of the song.
Resizing the window few pixels display the full song.

**Comment by tresf** [comment]
Possibly related #1490

**Comment by waxspin** [comment]
@tresf This issue appears to be fixed in the current master, testing on Kubuntu 15.10 64-bit.

**Comment by midi-pascal** [comment]
@waxspin @tresf Issue not fixed here (Ubuntu 14.04 64-bits on a freshly build master branch)

**Comment by zonkmachine** [comment]
Bug one was bad code formatting so it fixed it up a bit. The code in question is here:
https://github.com/LMMS/lmms/blob/master/src/core/main.cpp#L739:L807

Depending on what we load the following snippet is executed either before or after the loading action. No specific explanation is given to why there is a difference there.
```
gui->mainWindow()->show();
if( fullscreen )
{
	gui->mainWindow()->showMaximized();
}
```

The biggest difference between the different program starts is this. When you load or recover a file the show sequence is run first. When you import a file, open a recent project or create a new project the "show()" is run after.
I don't know why this is done differently.
Is this only on Ubuntu? There is a comment above the last occasion of the show() thing that indicates that this could be the case as the code is a workaround to appease FVWM2, a linux window manager.

```
// [Settel] workaround: showMaximized() doesn't work with
// FVWM2 unless the window is already visible -> show() first
```
I have Metacity and don't see this problem. I see a difference in how the projects load as with a recent file the project window is shown after the project is loaded and only the pop up messages and prompts is shown before while loading, but the end result is a correctly drawn screen.
```
$ wmctrl -m
Name: Metacity (Marco)
```

___

Solution:
I'll try and break out the recent file part and put it in a separate `if else{}` before the final else{} with the show() snippet in the beginning and that will take care of this specific problem.

**Comment by zonkmachine** [comment]
Fixed in #2555

**Comment by zonkmachine** [comment]
@waxspin @midi-pascal 

Could you please test if one of your projects that opens up buggy as recent project opens up correct if you set it as "Default Template" ? ( Menu/Save as default template ).
Settings - Reopen last project on start: Off
I think the factory default template may get away with being small and fast enough to load before some timer event kicks in while the project isn't done loading.
